### PR TITLE
Support properties configuration

### PR DESCRIPTION
**Pull request by Incarnation-p-lee**
* Support properties configuration from resources

* Add Unit test for gremlin propertiesConfiguration.

* Add 2 dependences for SpringJUnit

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -18,8 +18,10 @@
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
 
         <spring.springframework.version>5.0.1.RELEASE</spring.springframework.version>
-        <spring.boot.starter.test.version>2.0.0.RELEASE</spring.boot.starter.test.version>
         <spring.data.version>2.0.0.RELEASE</spring.data.version>
+        <spring.boot.version>2.0.0.RELEASE</spring.boot.version>
```
Comment (yungezz) [inline pom.xml]: are those spring boot version all under spring framework dependencies? can we import that dependencies management?

File: src/main/resources/application.properties [hunk 2]
```diff
@@ -0,0 +1,4 @@
+gremlin.endpoint=pli-gremlin-test.gremlin.cosmosdb.azure.com
```
Comment (yungezz) [inline src/main/resources/application.properties]: use some placeholder

File: pom.xml (hunk 1 above)
Comment (Incarnation-p-lee) [reply pom.xml]: thanks, dependencies management looks more clear here.

File: src/main/resources/application.properties [hunk 3]
```diff
@@ -0,0 +1,4 @@
+gremlin.endpoint=endpoint.gremlin.cosmosdb.azure.com
+gremlin.port=443
+gremlin.username=/dbs/database/colls/collection
```
Comment (sophiaso) [inline src/main/resources/application.properties:3]: This username is a collection?

**Review by sophiaso — APPROVED** [review]

File: src/main/resources/application.properties (hunk 3 above)
Comment (Incarnation-p-lee) [reply src/main/resources/application.properties:3]: yes, username consist of /dbs/${database}/colls/${collection} from gremlin side. Actually the string identified unique azure cosmosdb collection for all db APIs.

**Pull request merged**